server/src/main.rs

```diff
@@ -173,25 +155,26 @@ mod tests {
         run_server(DEFAULT_HOST, DEFAULT_PORT, Path::new(""));
     }
 
-    fn start_test_server() -> TestServer {
-        TestServer::build_with_state(|| {
+    fn start_test_server() -> TestServerRuntime {
+        TestServer::new(|| {
             let addr = GitRepos::new(git::load_repos(Path::new("test"))).start();
-            AppState {
-                git_repos: addr.clone(),
-            }
-        })
-        .start(|app| {
-            app.resource("/repo/{repo}", |r| r.h(&super::get_repo));
+
+            h1::H1Service::new(
```

Comment:
Expects an `actix-http` service to be passed, rather than the old API which expected `actix-web` types. This is fine, they are compatible anyway — it's just a shuffle around, really.